test/e2e/enable_api_group_versions_test.go

```diff
@@ -0,0 +1,580 @@
+					"leadSinger": "John Lennon",
+				},
+			},
+		},
+		{
+			name: "Latest common non-preferred supported version v2beta2 is restored (Priority 3, Case D).",
+			srcCRD: map[string]string{
+				"url":       "testdata/enable_api_group_versions/case-b-source-manually-added-mutations.yaml",
+				"namespace": "music-system",
+			},
+			srcCRs: map[string]string{
+				"v2beta2": "testdata/enable_api_group_versions/music_v2beta2_rockband.yaml",
+				"v2beta1": "testdata/enable_api_group_versions/music_v2beta1_rockband.yaml",
+				"v1":      "testdata/enable_api_group_versions/music_v1_rockband.yaml",
+			},
+			tgtCRD: map[string]string{
+				"url":       "testdata/enable_api_group_versions/case-d-target-manually-added-mutations.yaml",
+				"namespace": "music-system",
+			},
+			tgtVer: "v2beta2",
+			cm:     nil,
+			want: map[string]map[string]string{
+				"annotations": {
+					"rockbands.music.example.io/originalVersion": "v2beta2",
+				},
+				"specs": {
+					"leadSinger": "John Lennon",
+					"leadGuitar": "George Harrison",
+					"drummer":    "Ringo Starr",
+				},
+			},
+		},
+		{
+			name: "No common supported versions means no rockbands custom resource is restored.",
+			srcCRD: map[string]string{
+				"url":       "testdata/enable_api_group_versions/case-a-source.yaml",
+				"namespace": "music-system",
+			},
+			srcCRs: map[string]string{
+				"v1":       "testdata/enable_api_group_versions/music_v1_rockband.yaml",
+				"v1alpha1": "testdata/enable_api_group_versions/music_v1alpha1_rockband.yaml",
+			},
+			tgtCRD: map[string]string{
+				"url":       "testdata/enable_api_group_versions/case-b-target-manually-added-mutations.yaml",
+				"namespace": "music-system",
+			},
+			tgtVer: "",
+			cm:     nil,
+			want:   nil,
+		},
+		{
+			name: "User config map overrides Priority 3, Case D and restores v2beta1",
+			srcCRD: map[string]string{
+				"url":       "testdata/enable_api_group_versions/case-b-source-manually-added-mutations.yaml",
+				"namespace": "music-system",
+			},
+			srcCRs: map[string]string{
+				"v2beta2": "testdata/enable_api_group_versions/music_v2beta2_rockband.yaml",
+				"v2beta1": "testdata/enable_api_group_versions/music_v2beta1_rockband.yaml",
+				"v1":      "testdata/enable_api_group_versions/music_v1_rockband.yaml",
+			},
+			tgtCRD: map[string]string{
+				"url":       "testdata/enable_api_group_versions/case-d-target-manually-added-mutations.yaml",
+				"namespace": "music-system",
+			},
+			tgtVer: "v2beta1",
+			cm: builder.ForConfigMap("velero", "enableapigroupversions").Data(
+				"restoreResourcesVersionPriority",
+				`rockbands.music.example.io=v2beta1,v2beta2,v2`,
+			).Result(),
+			want: map[string]map[string]string{
+				"annotations": {
+					"rockbands.music.example.io/originalVersion": "v2beta1",
+				},
+				"specs": {
+					"leadSinger": "John Lennon",
+					"leadGuitar": "George Harrison",
+					"genre":      "60s rock",
+				},
+			},
+		},
+	}
+
+	for i, tc := range tests {
```

Comment:
Ideally these tests should use the Ginkgo `DescribeTable` function as we do in the [controller tests](https://github.com/vmware-tanzu/velero/blob/main/pkg/controller/server_status_request_controller_test.go#L61) as it ensures that each test case is reported individually. This can be improved in a follow up though and shouldn't block this PR.

Reply:
Thanks, @zubron! I'll try to divide the test cases into their own Describe blocks using the controller tests as my example. I'll have it up in a follow-up PR.